File: frontend/lib/features/auth/models/signup_request.dart
import 'package:equatable/equatable.dart';

class SignupRequest extends Equatable {
  final String email;
  final String password;
  final String name;
  final String? roll;
  final String? phoneNo;
  final String? roomNo;

  const SignupRequest({
    required this.email,
    required this.password,
    required this.name,
    this.roll,
    this.phoneNo,
    this.roomNo,
  });

  factory SignupRequest.fromJson(Map<String, dynamic> json) {
    return SignupRequest(
      email: json['email'] as String,
      password: json['password'] as String,
      name: json['name'] as String,
      roll: json['roll'] as String?,
      phoneNo: json['phoneNo'] as String?,
      roomNo: json['roomNo'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'password': password,
      'name': name,
      'roll': roll,
      'phoneNo': phoneNo,
      'roomNo': roomNo,
    };
  }

  @override
  List<Object?> get props => [email, password, name, roll, phoneNo, roomNo];
}

File: backend/src/main/java/dsi/ruet/backend/dto/auth/SignupRequest.java
package dsi.ruet.backend.dto.auth;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SignupRequest {
    private String email;           // Required - must exist in Users table
    private String password;        // Required - override placeholder
    private String name;            // Required - override placeholder
    private String roll;            // Optional - required only if role is STUDENT
    private String phoneNo;         // Optional - required only if role is STUDENT
    private String roomNo;          // Optional - for STUDENT role
}

